py-polars/src/conversion.rs

```diff
@@ -500,13 +537,61 @@ impl ToPyObject for Wrap<&DateChunked> {
     }
 }
 
+impl ToPyObject for Wrap<&DecimalChunked> {
+    fn to_object(&self, py: Python) -> PyObject {
+        let utils = UTILS.as_ref(py);
+        let convert = utils.getattr("_to_python_decimal").unwrap();
+        let py_scale = self.0.scale().to_object(py);
+        // if we don't know precision, the only safe bet is to set it to 39
```

Comment:
FMI: why is it `39` here and `38` elsewhere?

Reply:
Because, technically, Python's decimals are not restricted by prec=38 (unlike arrow decimals), and... technically we could have a number between `10.pow(38)` and `i128::MAX`? (we still need to decide whether we want to allow values in that range).

Reply:
Ah, I hadn't realized python Decimals didn't have this upper bound. My opinion would be to not support more than 38. Seems like it can only introduce edge cases and the value seems pretty low?

Reply:
Let's mark this as "to figure out". I'd lean towards allowing the whole i128 scale if prec=None, but you'll have to check it "at the edges" (only arrow edges though! since Python doesn't care).